yarn-project/types/src/abi/contract_artifact.ts

```diff
@@ -125,6 +125,14 @@ function generateFunctionArtifact(fn: NoirCompiledContractFunction): FunctionArt
   };
 }
 
+/**
+ * Returns true if the first parameter is kernel function inputs.
+ */
+function hasKernelFunctionInputs(params: ABIParameter[]): boolean {
+  const firstParam = params[0];
+  return firstParam?.type.kind === 'struct' && firstParam.type.path.includes('ContextInputs');
+}
+
```

Comment:
Not totally clear what this is doing and why it's necessary. Maybe just expand on the comment.

Reply:
The macros for private / public currently inject a parameter that is the ContextInputs, this just says if it is there